src/utilities/guards.ts

```diff
@@ -0,0 +1,27 @@
+export function isRecord(value: unknown): value is Record<PropertyKey, unknown> {
+  return typeof value === 'object' && value !== null
```

Comment:
this type would return `true` given `[]`, it would still fail when you check for `key in ...` but could be misused in the future?

Reply:
Good catch. Updated